grails-app/controllers/com/unifina/controller/api/KeyApiController.groovy

```diff
@@ -116,10 +116,10 @@ class KeyApiController {
 				res.removeFromKeys(key)
 			}
 
-			def query = Permission.where {
-				key == key
+			List<Key> keys = Permission.createCriteria().list {
+				eq("key", key)
 			}
-			query.deleteAll()
+			keys*.delete(flush: true)
```

Comment:
This does N queries, while old code does 1. Can we do this with 1 criteria query?